addons/controllers/clusterbootstrap_controller.go

```diff
@@ -167,7 +173,7 @@ func (r *ClusterBootstrapReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	if !cluster.GetDeletionTimestamp().IsZero() {
```

Comment:
Question: Do we need to add finalizer to the Cluster CR?

Reply:
since the clusterboostrap CR has a  reference pointing to the cluster, the line of thought is that we just need to add it to the clusterboostrap. Right now, the only thing we use the finalizer is to signal the machine-controller that the machines should be marked with the not delete pretermination hook. so we can get away with just adding it to the clusterboostrap CR. The only reason I can think of adding it to the cluster (at this time) would be if the clusterboostrap is deleted before the cluster and the machines, but I think that would be a "bug" in our logic.